catalog/resource_grants.go

```diff
@@ -290,6 +290,18 @@ func parseId(d *schema.ResourceData) (string, string, error) {
 	return split[0], split[1], nil
 }
 
+// Check whether it's a first pass of forced new. From CustomizeDiff in schema.Resource:
+// "Existing resource, forced new: One run with state (before ForceNew), then one run without state (as if new resource)"
+func firstPassOfForcedNew(d *schema.ResourceDiff) bool {
+	fieldSetAsId := strings.Split(d.Id(), "/")[0]
+	old, new := d.GetChange(fieldSetAsId)
+	if old.(string) != "" && new.(string) == "" {
+		// First pass of forced new
+		return true
+	}
+	return false
```

Comment:
This logic is incorrect, I believe. In the first pass, the state will be present, so there will be a change. It passes unit tests because our synthesized InstanceState has a bug: it doesn't include the resource ID in the state. This causes CustomizeDiff to be [skipped earlier](https://github.com/databricks/terraform-provider-databricks/pull/3163/files#diff-b368dac73336f7c30e2ef9bf3258af896959ba0b75433759c6bb3499fad4feedR325).

In this PR, let's update `testing.go` to have
```go
	is := &terraform.InstanceState{
		ID:         f.ID,
		Attributes: f.InstanceState,
	}
```
on line 302. Then, you can see that this check returns `false` for existing resources.